pandas/core/internals/blocks.py

```diff
@@ -621,9 +621,12 @@ def _astype(self, dtype, copy=False, errors="raise", **kwargs):
                 else:
                     values = self.get_values(dtype=dtype)
 
+                skipna = kwargs.pop("skipna", True)
                 # _astype_nansafe works fine with 1-d only
                 vals1d = values.ravel()
-                values = astype_nansafe(vals1d, dtype, copy=True, **kwargs)
+                values = astype_nansafe(
+                    vals1d, dtype, copy=True, skipna=skipna, **kwargs
```

Comment:
not sure I understand why you changed this code, is this not already passed thru?

Reply:
`astype` method doesn't have `skipna` keyword. `skipna` is passed in `kwargs`.
https://github.com/pandas-dev/pandas/blob/e0c63b4cfaa821dfe310f4a8a1f84929ced5f5bd/pandas/core/generic.py#L5764

`skipna` is pop out from `kwargs` if any. If not, `skipna=True` is set and passed to `astype_nansafe`. (In `astype_nansafe`, `skipna=False` is default.) 

https://github.com/pandas-dev/pandas/blob/e0c63b4cfaa821dfe310f4a8a1f84929ced5f5bd/pandas/core/dtypes/cast.py#L663

So, the default behaviour after this change is to skip converting `nan` into string.